src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTextUI.java

```diff
@@ -303,10 +303,8 @@ private void updateBackground(JTextComponent c) {
     /**
      * Initializes component properties, such as font, foreground,
      * background, caret color, selection color, selected text color,
-     * disabled text color, and border color.  The font, foreground, and
-     * background properties are only set if their current value is either null
-     * or a UIResource, other properties are set if the current
-     * value is null.
+     * disabled text color, and border color.  All properties are set
```

Comment:
This method is also affects a margin, but it is does not mentioned in the javadoc.

Reply:
Does the list of properties listed in the javadoc have to be comprehensive? However, `margin` seems to be the only property omitted from the list.

Reply:
I have added margin too,